Galaxy_OwncloudImportExport/owncloud_import.xml

```diff
@@ -0,0 +1,53 @@
+<tool id="owncloud_import" name="Import from owncloud" version="2.0">
+  <description>import dataset from your Owncloud/CloudStor account</description>
+
+  <environment_variables>
+    <environment_variable name="OWNCLOUD_SERVER_URL">
+      #set $server_url = $__user__.extra_preferences.get('owncloud_account|server_url', "")
+      #assert $server_url, Exception('Invalid server url. You must define your server url in your Galaxy user profile.')
+      $server_url
+    </environment_variable>
+    <environment_variable name="OWNCLOUD_USERNAME">
+      #set $username = $__user__.extra_preferences.get('owncloud_account|username', "")
+      #assert $username, Exception('Invalid username. You must define your username and password in your Galaxy user
+      profile.')
+      $username
+    </environment_variable>
+    <environment_variable name="OWNCLOUD_PASSWORD">
```

Comment:
might be nice to look into offering token-based authentication rather than only username/password? https://owncloud.org/news/introducing-oauth2-secure-authorization-flow/

Reply:
Works for both afaik, and token would be preferable, since we can revoke. The plan was to update the documentation to say that using the token is recommended.